### PR TITLE
Allow list of entity ids to be loaded to better support idpList

### PR DESCRIPTION
**Pull request by pradtke**
Fix for #1101
Certain metadata sources, like mdq, don't support loading all metadata which
prevented them from being used with idplist. This change adds support for allowing
metadata sources to load a list of entities, allowing mdq to be used for idp list.

### CODE REVIEW

File: lib/SimpleSAML/Metadata/MetaDataStorageHandler.php [hunk 1]
```diff
@@ -256,6 +257,38 @@ public function getPreferredEntityIdFromCIDRhint($set, $ip)
...
+    /**
+     * This function loads the metadata for entity IDs in $entityIds. It is returned as an associative array
+     * where the key is the entity id. An empty array may be returned if no matching entities were found
+     * @param array $entityIds The entity ids to load
+     * @param string $set The set we want to get metadata from.
+     * @return array An associative array with the metadata for the requested entities, if found.
+     */
+    public function getMetaDataForEntities(array $entityIds, $set)
+    {
+
```
Comment (tvdijen) [inline lib/SimpleSAML/Metadata/MetaDataStorageHandler.php]: This newline does not belong here as per PSR-2

File: lib/SimpleSAML/Metadata/MetaDataStorageSource.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public static function parseSources($sourcesConfig)
      *
      * @param array $sourceConfig Associative array with the configuration for this metadata source.
      *
-     * @return mixed An instance of a metadata source with the given configuration.
+     * @return MetaDataStorageSource An instance of a metadata source with the given configuration.
```
Comment (tvdijen) [inline lib/SimpleSAML/Metadata/MetaDataStorageSource.php]: I'd like to see the fully namespaced class name in phpdocs

**Comment by tvdijen** [comment]
Hi @pradtke ! Thanks for this, but unfortunately the Travis tests are failing.. I also left a few nitpicks ;)

File: lib/SimpleSAML/Metadata/MetaDataStorageHandler.php (hunk 1 above)
Comment (pradtke) [reply lib/SimpleSAML/Metadata/MetaDataStorageHandler.php]: Fixed

File: lib/SimpleSAML/Metadata/MetaDataStorageSource.php (hunk 2 above)
Comment (pradtke) [reply lib/SimpleSAML/Metadata/MetaDataStorageSource.php]: Fixed

**Pull request merged**